app/services/work_reminder_generator.rb

```diff
@@ -0,0 +1,40 @@
+# typed: false
+# frozen_string_literal: true
+
+# This creates and queues reminder emails for in-progress drafts
+class WorkReminderGenerator
+  extend T::Sig
+
+  # Sends the day's reminders about open first drafts, using default values for the notification
+  # interval unless the caller overrides the defaults with custom values. Intended to be run daily
+  # by a cron job, but the optional manual override for notification interval is useful in case a
+  # human has to manually call the method to send dropped notifications, e.g. in the event that
+  # something goes awry with the cron job on a particular day. For example, if the cron job got
+  # wedged 2 days ago, you could call this method with the the default values plus 2.
+  sig { params(first_interval: ActiveSupport::Duration, subsequent_interval: ActiveSupport::Duration).void }
+  def self.send_first_draft_reminders(first_interval: nil, subsequent_interval: nil)
+    first_interval ||= Settings.notifications.first_draft_reminder.first_interval.days
+    subsequent_interval ||= Settings.notifications.first_draft_reminder.subsequent_interval.days
+
+    Work.with_state(:first_draft, :version_draft).find_each do |work|
+      if send_first_draft_reminder?(work, first_interval, subsequent_interval)
+        WorksMailer.with(work: work)
+                   .first_draft_reminder_email.deliver_later
+      end
+    end
+  end
+
+  sig do
+    params(work: Work, first_interval: ActiveSupport::Duration, subsequent_interval: ActiveSupport::Duration)
+      .returns(T::Boolean)
+  end
+  def self.send_first_draft_reminder?(work, first_interval, subsequent_interval)
+    today = DateTime.now.utc.beginning_of_day
+    first_reminder_day = work.created_at.utc.beginning_of_day.to_datetime + first_interval
```

Comment:
Should this be `updated_at`?  Since we're looking at a time since they last touched this item.  Is there a reason we can't run this query in the database (e.g. show me the works that are in a draft state and haven't been touched for 2 weeks?)

Reply:
Hmm... so I was initially using `created_at` when I thought this only applied to the `:first_draft` state.  But additionally, I was also under the impression that the reminder interval should be based on when the work went _into_ the draft state, and not when it was last edited.

But looking back at the issue and the spec it links, that now seems ambiguous to me, and I don't think my assumption there was clearly founded.  @jcoyne or @amyehodge, could you clarify whether the desired behavior is for the notification interval to be based on how long the work has been in draft state, vs basing it on the last edit?  Even if it's the former (how long it's been in draft state), I think I could join to `events` to get that info, instead of looking at ~~`created_at`~~`updated_at`.  But, now that I'm not just looking at things in `:first_draft`, using `created_at` does clearly feel wrong no matter what.

@jcoyne, I had briefly considered limiting by date in the query, but I didn't have a strong preference about that when mulling it over.  After getting into the implementation, it seemed a bit easier to do the date math that would let me get everything from an entire given day in the Ruby code instead of in the query.  But it would also be easy enough to filter in the query with a cutoff that's one day later than it should be, and then to keep the same filtering in the Ruby code.  Or, I could just do the date math in SQL afterall 🙂   Do you have a strong preference on this front?

Reply:
oh actually, i suppose i could also just calculate the cutoff date in ruby code and pass that into the query.  so yeah, easy enough to do some date filtering in the query, if you'd prefer.

Reply:
It's not a big deal now, but I can imagine there might be thousands of objects in a draft states, so a sql query would prevent us from having to "fix" this down the road.

Reply:
yeah, i didn't have a great sense of what the expected scale would be, so i wasn't sure how important it'd be to optimize that part of things.  will fix the query.